test/StreamTest.php

```diff
@@ -68,6 +73,28 @@ public function testCanInstantiateWithStreamResource(): void
         $this->assertInstanceOf(Stream::class, $stream);
     }
 
+    public function testThatInstantiatingWithArbitraryStringCausesException(): void
+    {
+        $this->expectException(DiactorosRuntimeException::class);
+        $this->expectExceptionMessage('Invalid stream reference provided');
+        new Stream('foo');
+    }
+
+    public function testInvalidStringArgumentWithCustomErrorHandler(): void
+    {
+        set_error_handler(function (): never {
+            throw new Exception('Bad News');
+        }, E_WARNING);
+
+        try {
+            new Stream('foo');
+        } catch (DiactorosRuntimeException $error) {
+            self::assertStringContainsString('Invalid stream reference provided', $error->getMessage());
```

Comment:
Shouldn't we expect to see "Bad news" here?

Reply:
Probably previous exception? 🤔

Reply:
'Bad News' is swallowed by `src/Stream` try/catch - it should be there in `$error->previous`